File: backend/models/api/episode.go
package api

import dbModels "github.com/STREAM-BUSTER/stream-buster/models/db"

type Episode struct {
	MediaID uint
	Media   *dbModels.Media

	Name     string
	Overview string

	EpisodeTMDBID int
	EpisodeNumber int

	StillPath string
	Runtime   int

	SeasonNumber int
}

File: frontend/src/models/episode.ts
import { Media } from './media';

export interface Episode {
  MediaID: number;
  Media?: Media;  // Optional field
  Name: string;
  Overview: string;
  EpisodeTMDBID: string;
  EpisodeNumber: number;
  StillPath: string;
  Runtime: number;
  SeasonNumber: number;
}